doc/partitioned-db.md

```diff
@@ -0,0 +1,203 @@
+# Partitioned Database
+
+When creating a Cloudant database, you can configure it as non-partitioned or partitioned.
+A partitioned database offers significant performance and cost advantages but requires you to specify a logical partitioning of your data.
+
+To get familiar with this feature, you can read the following references as the background of our new epic:
+
+- The Cloudant official document for database partitioning: https://cloud.ibm.com/docs/services/Cloudant/guides?topic=cloudant-database-partitioning
+- The related APIs' document from our driver module `nodejs-cloudant`: https://github.com/cloudant/nodejs-cloudant#partitioned-databases
+
+Now let's start with a user scenario to understand what's been added in the new feature.
```

Comment:
should this line be right above the paragraph begins with "Suppose you have a Product model..."?

Reply:
Is it better to mention here that the user scenario you showed below will show how to:
- add partitioned index
- create document in Partitioned Database
- define partitioned field
- perform partitioned find